### ISSUE TITLE
Possible build error for v0.10.0

### ISSUE THREAD

**Issue by ijlee2** [ember-animation/ember-animated#190]
**Describe the bug**
I have a tutorial app written in Ember 3.14. After installing Ember Animated v0.10, I get a build error when I run `ember s`.

```
Build Error (broccoli-persistent-filter:Babel > [Babel: ember-animated]) in ember-animated/-private/sprite.ts

/<directory>/embercasts-library-ui/ember-animated/-private/sprite.ts: Unexpected token, expected "{" (472:36)

  470 |   }
  471 | 
> 472 |   assertHasInitialBounds(): asserts this is SpriteWithInitialBounds {
      |                                     ^
  473 |     if (!this._initialBounds) {
  474 |       throw new Error(`sprite does not have initialBounds`);
  475 |     }

```

By downgrading Ember Animated to v0.9, I am able to run the app again.

**To Reproduce**
I've isolated the app to a new repo for another bug report: https://github.com/ijlee2/bug-upgrade-313-to-314.

I believe you can clone this repo, install Ember Animated v0.10, then run `ember s`. You can encounter the error described above with either Ember 3.13 or 3.14.

**Expected Behavior**
I'm not sure if the error is due to a configuration that I might have made or another addon. Installing Ember Animated v0.10 on a new Ember 3.14 app didn't seem to run into a build issue.

**Screenshots**
Please see the error message above.

**Desktop (please complete the following information):**
 - OS: Mac 10.15
 - Browser: Chrome
 - Version 78

**Smartphone (please complete the following information):**
N/A

**Additional context**
N/A

**Comment by ef4** [comment]
I think this is caused by https://github.com/typed-ember/ember-cli-typescript/issues/994

**Comment by ef4** [comment]
Here is a program that will print out a report, save it to `trace.js` and run `node trace.js` in your ember app's directory:

```js
const { dirname, join } = require('path');
const { chdir, cwd } = require('process');
const { sync: resolveSync } = require('resolve');
const { satisfies, coerce } = require('semver');

const suspects = ['ember-animated', 'ember-cli-typescript', 'ember-cli-babel', '@babel/core', '@babel/parser'];

let chain = [];
let dir = cwd();

let pkg = require(join(dir, 'package.json'));
chain.push({ dir, pkg });

for (let name of suspects) {
  dir = dirname(resolveSync(name + '/package.json', { basedir: dir }));
  let pkg = require(join(dir, 'package.json'));
  chain.push({ dir, pkg });
}

let table = [];
for (let i = 0; i < chain.length - 1; i++) {
  let entry = chain[i];
  let wanted = Object.assign({}, entry.pkg.dependencies, entry.pkg.devDependencies)[suspects[i]];
  let have = chain[i + 1].pkg.version;
  let status = satisfies(coerce(have), wanted) ? 'GOOD' : 'BAD';
  table.push({ dep: suspects[i], wanted, have, status });
}
console.table(table);
```

**Comment by ef4** [comment]
Basically, this is probably a yarn and/or NPM bug. Packages are not getting the versions of their dependencies that they asked for.

**Comment by ijlee2** [comment]
Ok, sounds good. Thank you for investigating!

**Comment by alexlafroscia** [comment]
I was able to resolve this issue by doing some manual editing to my `yarn.lock` (which I preferred, compared to blowing the whole thing away).

Basically, I went through every `@babel/*` package in my `yarn.lock` and manually made each entry resolve to the latest version of that package that I had installed. For example, if I had something like

```
"@babel/foobar^1.0.0":
  "version": "1.0.0"
  "blah": "blah"

"@babel/foobar^1.0.1":
  "version": "1.0.1"
  "blah": "blah"
```

I went and replaced it with

```
"@babel/foobar^1.0.0", "@babel/foobar^1.0.1":
  "version": "1.0.1"
  "blah": "blah"
```

This is essentially what would have happened if I blew away the whole `yarn.lock` but restricted to _just_ the "@babel/*" dependencies.
